src/Illuminate/Foundation/Exceptions/Renderer/Listener.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Illuminate\Foundation\Exceptions\Renderer;
+
+use Illuminate\Contracts\Events\Dispatcher;
+use Illuminate\Database\Events\QueryExecuted;
+use Laravel\Octane\Events\RequestReceived;
+use Laravel\Octane\Events\RequestTerminated;
+use Laravel\Octane\Events\TaskReceived;
+use Laravel\Octane\Events\TickReceived;
+
+class Listener
+{
+    /**
+     * The queries that have been executed.
+     *
+     * @var array<int, array{connectionName: string, time: float, sql: string, bindings: array}>
+     */
+    protected $queries = [];
+
+    /**
+     * Register the appropriate listeners on the given event dispatcher.
+     *
+     * @param  \Illuminate\Contracts\Events\Dispatcher  $events
+     * @return void
+     */
+    public function registerListeners(Dispatcher $events)
+    {
+        $events->listen(QueryExecuted::class, [$this, 'onQueryExecuted']);
+
+        if (isset($_SERVER['LARAVEL_OCTANE'])) {
+            $events->listen([RequestReceived::class, TaskReceived::class, TickReceived::class, RequestTerminated::class], function () {
```

Comment:
@nunomaduro does this need to also reset on any job events?

Reply:
I've replicated what ignition is doing: https://github.com/spatie/laravel-ignition/blob/9c60d929a91803dee7e47614c37cbe6d3f2b1e2c/src/IgnitionServiceProvider.php#L311C1-L311C43.